cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -538,6 +539,550 @@ void accumulate_edge_results(
+      vertex_last, thrust::make_counting_iterator(origin_t{static_cast<uint32_t>(num_sources)}));
+
+    // Insert tagged sources into frontier
+    vertex_frontier.bucket(bucket_idx_cur).insert(pair_first, pair_last);
+
+    // Initialize distances and sigmas for sources
+    thrust::for_each(handle.get_thrust_policy(),
+                     pair_first,
+                     pair_last,
+                     [d_sigma_2d    = sigmas_2d.begin(),
+                      d_distance_2d = distances_2d.begin(),
+                      vertex_partition,
+                      num_vertices] __device__(auto tagged_source) {
+                       auto v      = thrust::get<0>(tagged_source);
+                       auto origin = thrust::get<1>(tagged_source);
+                       auto offset =
+                         vertex_partition.local_vertex_partition_offset_from_vertex_nocheck(v);
+                       auto idx           = origin * num_vertices + offset;
+                       d_distance_2d[idx] = 0;
+                       d_sigma_2d[idx]    = 1;
+                     });
+  }
+
+  edge_t hop{0};
+
+  while (vertex_frontier.bucket(bucket_idx_cur).aggregate_size() > 0) {
+    // Step 1: Extract ALL edges from frontier (filtered by unvisited vertices)
+    using bfs_edge_tuple_t = thrust::tuple<vertex_t, origin_t, edge_t>;
+
+    auto result = extract_transform_if_v_frontier_outgoing_e(
+      handle,
+      graph_view,
+      vertex_frontier.bucket(bucket_idx_cur),
+      edge_src_dummy_property_t{}.view(),
+      edge_dst_dummy_property_t{}.view(),
+      edge_dummy_property_t{}.view(),
+      cuda::proclaim_return_type<bfs_edge_tuple_t>(
+        [d_sigma_2d = sigmas_2d.begin(), num_vertices, vertex_partition] __device__(
+          auto tagged_src, auto dst, auto, auto, auto) {
+          auto src        = thrust::get<0>(tagged_src);
+          auto origin     = thrust::get<1>(tagged_src);
+          auto src_offset = vertex_partition.local_vertex_partition_offset_from_vertex_nocheck(src);
+          auto src_idx    = origin * num_vertices + src_offset;
+          auto src_sigma  = static_cast<edge_t>(d_sigma_2d[src_idx]);
+
+          return thrust::make_tuple(dst, origin, src_sigma);
+        }),
+      // PREDICATE: only process edges to unvisited vertices
+      cuda::proclaim_return_type<bool>(
+        [d_distances_2d = distances_2d.begin(),
+         num_vertices,
+         vertex_partition,
+         invalid_distance] __device__(auto tagged_src, auto dst, auto, auto, auto) {
+          auto origin     = thrust::get<1>(tagged_src);
+          auto dst_offset = vertex_partition.local_vertex_partition_offset_from_vertex_nocheck(dst);
+          auto dst_idx    = origin * num_vertices + dst_offset;
+          return d_distances_2d[dst_idx] == invalid_distance;
+        }));
+
+    // Step 2: Work directly with the dataframe buffer components (no temporaries needed)
+    auto new_frontier_tagged_vertex_buffer = std::move(result);
```

Comment:
Just name `result` to `new_frontier_tagged_vertex_buffer`. No need for this line.